src/app/features/collectibles/ordinals.tsx

```diff
@@ -1,89 +1,91 @@
-  //     </>
-  //   );
-  // }
+function Inscription({ path: id }: InscriptionProps) {
+  const { isLoading, isError, data } = useInscriptionQuery(id);
+
+  if (isLoading) return null; // TODO
+
+  if (isError) return null; // TODO
+
+  const ordinal = whenOrdinalType(data['content type'], {
+    image: () => ({
+      type: 'image',
+      title: data.title,
+      infoUrl: createInfoUrl(data.content),
+      content: `https://ordinals.com${data.content}`,
+    }),
+    other: () => ({
+      type: 'other',
+      title: data.title,
+      infoUrl: `https://ordinals.com${data.content}`.replace('content', 'inscription'),
+    }),
+  });
 
-  // if (Math.random() < 1) {
-  //   return (
-  //     <OtherCollectible
-  //       onClick={() => alert('clicked')}
-  //       subtitle="Ordinal inscription"
-  //       title="Foo"
-  //     />
-  //   );
-  // }
+  switch (ordinal.type) {
+    case 'image': {
+      return (
+        <ImageCollectible
+          key={ordinal.infoUrl}
+          onClick={() => openInNewTab(ordinal.infoUrl)}
+          src={ordinal.content}
+          subtitle="Ordinal inscription"
+          title={ordinal.title}
+        />
+      );
+    }
+    case 'other': {
+      return (
+        <OtherCollectible
+          key={ordinal.infoUrl}
+          onClick={() => openInNewTab(ordinal.infoUrl)}
+          subtitle="Ordinal inscription"
+          title={ordinal.title}
+        />
+      );
+    }
+    default: {
+      return null;
+    }
+  }
+}
+
+interface TransactionQueryLoaderProps {
+  q: UseQueryResult<OrdApiXyzGetTransactionOutput>;
+}
+function TransactionQueryLoader({ q }: TransactionQueryLoaderProps) {
+  if (q.isLoading) return null; // TODO
 
-  if (isLoading) return null;
+  if (q.isError) return null; // TODO
 
-  // TODO: user facing error handling.
-  if (isError) return null;
+  return <Inscription path={q.data.inscriptions} />;
+}
 
-  if (data.length === 0) return null;
+export function Ordinals() {
+  const { data: utxos } = useTaprootAddressUtxosQuery();
+  const transactionsMetadata = useTransactionsMetadataQuery(utxos ?? []);
 
   return (
     <>
-      {data.map(ordinal => {
-        switch (ordinal.type) {
-          case OrdinalType.Image: {
-            return (
-              <ImageCollectible
-                onClick={() => openInNewTab(ordinal.infoUrl)}
-                src={ordinal.content}
-                subtitle="Ordinal inscription"
-                title={ordinal.title}
-              />
-            );
-          }
-          case OrdinalType.Other: {
-            return (
-              <OtherCollectible
-                onClick={() => openInNewTab(ordinal.infoUrl)}
-                subtitle="Ordinal inscription"
-                title={ordinal.title}
-              />
-            );
-          }
-          default: {
-            return null;
-          }
-        }
-      })}
+      {transactionsMetadata.map((q, i) => (
+        <TransactionQueryLoader key={i} q={q} />
```

Comment:
Not quite sure what to use as a key here. Using the index isn't best practice, but the query object doesn't seem to have anything uniquely identifiable about it. Suggestions?

Reply:
If there's no ID or nothing unique, I don't see any alternative to the index, even if bad practice